index.js

```diff
@@ -1,33 +1,59 @@
 require('dotenv').config();
 
 const fetch = require('node-fetch');
-const unzipper = require('unzipper');
+const unzipper = require('unzip-stream');
+const path = require('path');
 
 const { DATA_URL } = require('./config/download');
 const { getTable, insertData } = require('./bigquery');
 const bqConfig = require('./config/bigquery');
+const Downloader = require('nodejs-file-downloader');
 
 const downloadFile = async () => {
   const res = await fetch(DATA_URL);
   const json = await res.json();
   const url = json.result.resources.map(r => r.url).find(url => url.includes('full'));
 
-  const fileReq = await fetch(url);
-  fileReq.body.pipe(unzipper.Parse())
-    .on('entry', function (entry) {
-      const tables = ['uo', 'fop'];
+  let prevPercent = 0;
+  const downloader = new Downloader({
+    url,
+    directory: './downloads',
+    onProgress: (percentage) => {
+      if (Math.floor(percentage) !== prevPercent) {
+        prevPercent = Math.floor(percentage);
+        console.log(percentage, '%');
+      }
+    },
+  });
+  try {
+    await downloader.download();
+    console.log('Download finished!');
 
-      console.log('got file', entry.path);
-      const table = tables.find(table => entry.path.toLowerCase().includes(table));
+    const filename = url.split('/').pop();
+    require('fs').createReadStream('./downloads/' + filename).pipe(unzipper.Parse())
+      .on('entry', function (entry) {
```

Comment:
for await?
https://habr.com/ru/post/353886/

Reply:
Not sure if it's applicable here, because `entry` event doesn't emit file chunks, but emits file entries in the archive that could be further drained or piped to the parser pipeline

Reply:
contract is similar, it should work